### PR TITLE
added option to disable flatten nested

### PR DESCRIPTION
**Pull request by tmahany419**
##### SUMMARY
I could not find a way to run multiple queries from the same context using this module. There's no way to create a table with `flatten_nested` disabled without running multiple queries. This PR adds a field called `flatten_nested` for `clickhouse_client` that sets this setting before running the query. 

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME

Flatten Nested

##### ADDITIONAL INFORMATION

This is clickhouse's document explaining options for nested fields: https://clickhouse.com/docs/en/sql-reference/data-types/nested-data-structures/nested. The behavior with `flatten_nested` enabled and disabled is explained there. The only way to disable it is by running `set flatten_nested=0;` before creating a table.

### CODE REVIEW

File: plugins/doc_fragments/client_inst_opts.py [hunk 1]
```diff
@@ -49,6 +49,14 @@ class ModuleDocFragment(object):
...
     default: {}
 
+  flatten_nested:
+    description:
+      - Updates flatten_nested setting on session before running
+        query
+      - Only runs the update if set to 0
+    type: int
+    default: 1
+
```
Comment (Andersson007) [inline plugins/doc_fragments/client_inst_opts.py:59]:
```suggestion
  flatten_nested:
    description:
      - Sets the C(flatten_nested) setting on session before running the query.
    type: int
    choices: [0, 1]
    version_added: '0.5.0'
```
- this is the `clickhouse_client` related argument, so please move it to plugins/modules/clickhouse_client.py
- let's skip the default and let users specify it explicitly?

File: plugins/module_utils/clickhouse.py [hunk 2]
```diff
@@ -37,6 +37,7 @@ def client_common_argument_spec():
         login_user=dict(type='str', default=None),
         login_password=dict(type='str', default=None, no_log=True),
         client_kwargs=dict(type='dict', default={}),
+        flatten_nested=dict(type='int', default=1),
```
Comment (Andersson007) [inline plugins/module_utils/clickhouse.py:40]:
```suggestion
        flatten_nested=dict(type='int'),
```
please move this to plugins/modules/clickhouse_client.py too

File: plugins/modules/clickhouse_client.py [hunk 3]
```diff
@@ -283,6 +284,8 @@ def main():
     substituted_query = get_substituted_query(module, client, query, execute_kwargs)
 
     # Execute query
+    if flatten_nested == 0:
+        execute_query(module, client, "set flatten_nested=0;", execute_kwargs)
```
Comment (Andersson007) [inline plugins/modules/clickhouse_client.py:288]:
```suggestion
        execute_query(module, client, "SET flatten_nested = 0", execute_kwargs)
```

**Review by Andersson007 — CHANGES_REQUESTED** [review]
@tmahany419 hello, thanks for the PR!
Could you also please:
- Add a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment) 
- Add a test task to https://github.com/ansible-collections/community.clickhouse/blob/main/tests/integration/targets/clickhouse_client/tasks/initial.yml

thanks

**Comment by tmahany419** [comment]
@Andersson007 Thank you for the feedback! I committed the changes you proposed. I've made unit test and it works against a local clickhouse db. I added it to the file you linked in the branch I am working on. Is there a way to test it before updating the PR?

**Comment by Andersson007** [comment]
> @Andersson007 Thank you for the feedback! I committed the changes you proposed. I've made unit test and it works against a local clickhouse db. I added it to the file you linked in the branch I am working on. Is there a way to test it before updating the PR?

@tmahany419 thanks!
About testing, yes, you can run the tests locally before pushing, please take a look at https://docs.ansible.com/ansible/devel/community/create_pr_quick_start.html , there are corresponding sections there

File: plugins/module_utils/clickhouse.py [hunk 4]
```diff
@@ -37,6 +37,7 @@ def client_common_argument_spec():
         login_user=dict(type='str', default=None),
         login_password=dict(type='str', default=None, no_log=True),
         client_kwargs=dict(type='dict', default={}),
+        flatten_nested=dict(type='int'),
```
Comment (Andersson007) [inline plugins/module_utils/clickhouse.py:40]:
```suggestion
        flatten_nested=dict(type='int', choices=[0, 1]),
```

**Comment by Andersson007** [comment]
@tmahany419 I don't see the tests, maybe you forgot to add a change to your commit?

**Comment by tmahany419** [comment]
@Andersson007 I updated the wrong branch yesterday, it should be there now.

**Comment by aleksvagachev** [comment]
@Andersson007 Maybe it is more correct to implement the transfer of not a specific parameter, but the desired list of parameters?

**Comment by Andersson007** [comment]
@aleksvagachev good catch, something like the following?
```
- name: Run the query
  community.clickhouse.clickhouse_client:
  query: SELECT ...
  set:
    flatten_nested: 0
```
?

**Comment by aleksvagachev** [comment]
@Andersson007 Yes, something similar. Shall we try to implement this in 1.0.0?

**Comment by Andersson007** [comment]
> @Andersson007 Yes, something similar. Shall we try to implement this in 1.0.0?

@aleksvagachev as this change hasn't been released yet, we can do it right now. Thoughts? Would you like to take it?

**Comment by aleksvagachev** [comment]
@Andersson007 I'll try to do it, but not before Monday

**Pull request merged**